codec/src/main/java/io/netty/handler/codec/compression/BrotliDecoder.java

```diff
@@ -0,0 +1,29 @@
+package io.netty.handler.codec.compression;
+
+import com.nixxcode.jvmbrotli.dec.BrotliInputStream;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.buffer.ByteBufOutputStream;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.ByteToMessageDecoder;
+
+import java.util.List;
+
+public class BrotliDecoder extends ByteToMessageDecoder {
+
+    @Override
+    protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) throws Exception {
+        ByteBuf decompressed = ctx.alloc().buffer();
+
+        ByteBufOutputStream byteBufOutputStream = new ByteBufOutputStream(decompressed);
+
+        BrotliInputStream decompressorIS = new BrotliInputStream(new ByteBufInputStream(in));
+        int read = decompressorIS.read();
+        while(read > -1) {
```

Comment:
I don't think this decode is meant to assume it's decoding a whole message necessarily.  You might need to change allocation of the BrotliInputStream, so it's not 1:1 with calls to decode. 

See state management in https://github.com/netty/netty/blob/5ea476a890bc367505e086f34c91d90f26411072/codec/src/main/java/io/netty/handler/codec/compression/JZlibDecoder.java for example

Reply:
I will leave the Decoder for later. Can you see Encoder?

Reply:
Finished Decoder. PTAL.